### PR TITLE
Add files via upload

### PR DESCRIPTION
**Pull request by polinavino**
Some verbal descriptions of delegation rules

### CODE REVIEW

File: latex/delegation.tex [hunk 1]
```diff
@@ -166,13 +198,39 @@
   %
   \begin{equation*}
     \_ \vdash \_ \trans{pool}{\_} \_ \in
-      \powerset (\Epoch \times \DState \times \DCert \times \DState)
+      \powerset (\Epoch \times \PState \times \DCert \times \PState)
```
Comment (JaredCorduan) [inline latex/delegation.tex]: good catch!

File: latex/delegation.tex [hunk 2]
```diff
@@ -253,6 +311,46 @@
...
+
+The rules for registering and delegating stake keys are given in
+\cref{fig:delegation-rules}. The preconditions for the registration of a stake
+key ensure that a given certificate $c$ is of the correct type
+(i.e. $\DCertRegKey$),
+and that the hash key associated with the author of the certificate is not
+already found in the current list of stake keys.
+
+In order to delegate to a stake pool, a stake key must first be registered.
+When registering a new stake key (the Deleg-Reg inference rule),
```
Comment (JaredCorduan) [inline latex/delegation.tex]:
```suggestion
When registering a new stake key (the \cref{eq:deleg-reg} inference rule),
```

File: latex/delegation.tex [hunk 3]
```diff
@@ -55,14 +55,13 @@
 Delegation requires the following to be tracked by the ledger state:
 the registered stake keys, the delegation map from registered stake keys to stake
 pools, the registered stake pools, and upcoming stake pool retirements.
-Additionally, the blockchain protocol rewards eligible stake, and so we must
-also include a mapping from active stake keys to rewards.
+Delegation, registration and deregistration are all performed via certificates.
+These certificates need not be kept track of in any way after they are validated
+and applied (i.e. the action it certifies is performed).
```
[the remark this answers is not in the document]
Comment (dnadales) [reply latex/delegation.tex]: I agree, I think it might be worthwhile to clarify what happens with pool registration certificates.

File: latex/delegation.tex [hunk 4]
```diff
@@ -333,6 +431,39 @@
...
+and less than $\emax$ epochs after the current one, and also that
+the certificate author's key is in the list of registered pools.
+This rule simply adds the pair of the key of the
+certificate author and the planned retirement epoch $e$ to the $retiring$
+parameter.
+
+The third rule, Pool-Reap, is used when the planned retirement epoch $e$ for
+a given stake pool in the $\fun{retiring}$ set is reached. In this case, the
+pair associated with the $stpools$
+
```
Comment (mgudemann) [inline latex/delegation.tex]: Somehing seems to be missing here

File: latex/delegation.tex [hunk 5]
```diff
@@ -398,4 +529,3 @@ \subsection{Witnesses}
   \caption{Delegation witnesses inference rules}
   \label{fig:rules:delegationw}
 \end{figure}
-
```
Comment (mgudemann) [inline latex/delegation.tex]: unnecessary change

File: latex/delegation.tex [hunk 6]
```diff
@@ -55,14 +55,13 @@
...
-Additionally, the blockchain protocol rewards eligible stake, and so we must
-also include a mapping from active stake keys to rewards.
+Delegation, registration and deregistration are all performed via certificates.
+These certificates need not be kept track of in any way after they are validated
+and applied (i.e. the action it certifies is performed).
 
-In \cref{fig:delegation-definitons} we give the delegation primitives,
-and in \cref{fig:delegation-transitions} we give the delegation transition rule.
+Additionally, the blockchain protocol rewards eligible stake, and so in the
+future we will also include a mapping from active stake keys to rewards.
```
[the remark this answers is not in the document]
Comment (polinavino) [reply latex/delegation.tex]: I commented this out. I will talk more about this in my next LaTeX PR once I am familiar with the mapping.

File: latex/delegation.tex (hunk 3 above)
[the remark this answers is not in the document]
Comment (polinavino) [reply latex/delegation.tex]: I added a note about this
\begin{note}
  Pool registration certificate data is kept as long as the pool is registered.
  Should this be part of the ledger state in some way?
\end{note}

File: latex/delegation.tex [hunk 7]
```diff
@@ -134,6 +135,37 @@
   \label{fig:delegation-definitons}
 \end{figure}
 
+
+In \cref{fig:delegation-definitons} we give the delegation primitives.
+The type $\DCert$ is a generic certificate type, which can be a registration,
+deregistration or delegation certificate for a key, or a registration/retirement
+ certificate for a stake pool. It is denoted as disjoint union in the figure,
+one should, however, think of it as a type with five initial constructors, one for
```
[the remark this answers is not in the document]
Comment (polinavino) [reply latex/delegation.tex]: I said instead :
It is denoted as disjoint union in the figure,
one should, however, think of a term of this type as a term of a specific
one of these five subtypes.

File: latex/delegation.tex (hunk 2 above)
Comment (polinavino) [reply latex/delegation.tex]: I changed all the rule names to references

File: latex/delegation.tex (hunk 4 above)
Comment (polinavino) [reply latex/delegation.tex]: Fixed the last sentence:
In this case, all the
pairs associated with hash keys of pools which are scheduled to retire in this
epoch are removed
from both $\var{stpools}$ and $\var{retiring}$.

And added a note:
\begin{note}
$\var{retired}$ and $\var{retiring}$ are both finite maps. For domain exclusion,
we only need the hashes of $\var{retired}$?
\end{note}

File: latex/delegation.tex [hunk 8]
```diff
@@ -166,13 +198,39 @@
...
+state transition types. These transitions can be thought of as collections
+ of maps between states indexed by certificates (as well the epoch
+context, in the case of stake pools).
+The one notable
+difference between the types of these transitions is the presence of
+$cepoch$ in the context of the stake pool transition.
+
+The part of the ledger state keeping track of current delegations, $\DState$,
+consists of a set of registered stake keys (the hash keys of those addresses
+which delegete to some stake pool), rewards, which are expressed as a finite
```
[the remark this answers is not in the document]
Comment (polinavino) [reply latex/delegation.tex]: Oops! Fixed that just now, will upload in the next PR

**Comment by JaredCorduan** [comment]
closing this PR since the information is now in #34
